newsroom/template_filters.py

```diff
@@ -61,6 +62,78 @@ def parse_date(datetime):
     return datetime
 
 
+def get_schedule_type(start, end, all_day, no_end_time):
+    duration = int((end - start).total_seconds() / 60)  # Convert seconds to minutes
+    DAY_IN_MINUTES = 24 * 60 - 1
+
+    if all_day:
+        return "ALL_DAY" if duration == 0 else "MULTI_DAY"
+
+    if no_end_time:
+        return "NO_DURATION"
+
+    if duration > DAY_IN_MINUTES or start.date() == end.date():
+        return "MULTI_DAY"
+
+    if duration == DAY_IN_MINUTES and start.date() == end.date():
+        return "ALL_DAY"
+
+    if duration == 0 and start.time() == end.time():
+        return "NO_DURATION"
+
+    return "REGULAR"
+
+
+def is_item_tbc(item):
+    event_tbc = item.get("event", {}).get("_time_to_be_confirmed", False)
+    if not event_tbc:
+        planning = item.get("planning_items", [])
+        return planning[0].get("_time_to_be_confirmed", False) if planning else False
+    return event_tbc
+
+
+def format_event_datetime(item):
+    datetime = item.get("dates", {})
+    if datetime:
+        tz = datetime.get("tz", get_session_timezone())
+        try:
+            # Set the session timezone
+            set_session_timezone(tz)
```

Comment:
I'm not sure but afaik babel caches it so you might have to call refresh after changing it

Reply:
I checked but IMO, we don't need to call refresh